cpp/models/lct_secir/infection_state.h

```diff
@@ -175,24 +193,24 @@ class InfectionState
      */
     bool check_constraints() const
```

Comment:
Something else I've just noticed. We have the public function check_constraints() here. In itself, I think the naming is good. But this could be a bit confusing with the existing check_constraints() functions for the parameters and population. I would recommend making the name more specific.

Reply:
My idea here was that the function should be similar to the ones from parameters and population. It actually has the same purpose, except that I call it in a different place to avoid calculations with invalid vectors.